### ISSUE TITLE
[Platform] Create/Edit universe form redesign

### ISSUE THREAD

**Issue by ramkumarvs** [yugabyte/yugabyte-db#4578]
Update our Create/Edit universe form to match the new design.
https://xd.adobe.com/view/87633f0c-d331-44a6-aa2b-72c9a853a3f5-d230/

Implementation sub-steps:
✅ Create new components required for the new design
✅ Cloud Config step
✅ Instance Config step
✅ DB Config step
✅ Security Config step
✅ Review step
⬜️ Summary widget (https://github.com/yugabyte/yugabyte-db/issues/6379, https://github.com/yugabyte/yugabyte-db/issues/6437)
⬜️ Misc (https://github.com/yugabyte/yugabyte-db/issues/6438, https://github.com/yugabyte/yugabyte-db/issues/6439, https://github.com/yugabyte/yugabyte-db/issues/6440)

Aha! Link: https://yugabyte-test.aha.io/features/PLATFORM-396

**Comment by sshev** [comment]
Created new components required for the new design, working on form implementation.

**Comment by Grishshah** [comment]
Widget flow mockup: https://xd.adobe.com/view/9a324f0d-cbfe-47bb-5181-d788355242e1-ec03/

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-yb-issue-4578-analysis.md`):

# Analysis: yugabyte/yugabyte-db#4578 (no Jira link; Aha PLATFORM-396)

Issue: "[Platform] Create/Edit universe form redesign" (filed 2020-05-26 by ramkumarvs, area/platform +
area/ui, milestone 2.5.x, OPEN, last updated 2020-11-20, 2 comments). An implementation tracker for
updating the YugabyteDB Anywhere (YBA, `managed/ui/`) Create/Edit universe form to a 2020 Adobe XD design.
Checked sub-steps: new components, Cloud Config, Instance Config, DB Config, Security Config, Review step.
Unchecked: Summary widget (#6379, #6437) and Misc (#6438, #6439, #6440). Component: YBA UI (`managed/ui/`).
Kind: feature/redesign tracker; validated as a behavior delta (what the tracker asked for vs. what the
current form ships), not as a crash reproduction. Validated 2026-09-05.

## Verdict
NOT-REPRODUCED -- the historical gap ("form does not match the new design") no longer exists on any
required target: the #4578 wizard was delivered (2020-07..2020-11, shipped v2.3.x-v2.17.2.0 behind a
"Try New UI" link), then replaced by the PLAT-7179/PLAT-7545 redesigned form that is the default on every
in-support branch and master, and by the features-v2 wizard (2025.2+, 2026.1, master). The unchecked
sub-steps are delivered by those successors except two small confirmation dialogs from #6438
(leave-unsaved-form, Reset Config), which remain unimplemented everywhere and are tracked by #6438 itself.

## Root cause
Not a defect; a superseded design tracker. Mechanism of the "gap" and its closure:

- The #4578 deliverable was the wizard at `managed/ui/src/redesign/universe/wizard/` (steps
  `UniverseWizard.tsx:177-181` at c1fda7ffe3: CloudConfig, InstanceConfig, DBConfig, SecurityConfig,
  Review), routed at `/universe/create` and `/universe/:universeId/edit/primary` (`routes.js:143,145` at
  c1fda7ffe3) and reachable only through the legacy form's "Try New UI" link
  (`components/universes/UniverseForm/UniverseForm.js:719` at v2.16.0.0). It was never the default.
- The unchecked "Summary widget" was a stub: `wizard/compounds/Summary/Summary.tsx:36` at c1fda7ffe3 reads
  `// TODO: trigger "/universe_resources" api on form data change` and renders hard-coded `0` cores/GB;
  #6379 explains why (`/universe_resources` needed a `nodeDetailsSet` the wizard never produced).
- PLAT-7179 (ac66da2c15, 2023-01-30) merged the `universe_code_revamp` redesign at
  `managed/ui/src/redesign/features/universe/universe-form/`; PLAT-7545 (df6efc876f, 2023-03-01) made it
  the default (`reducers/feature.js:27` `enableNewUI: true` on master; `pages/UniverseNewView.jsx:7-9`
  routes `/universes/create` (`routes.jsx:286`) to `UniverseFormContainer`); PLAT-7712 (f185c02e87,
  2023-03-14) deleted the #4578 wizard.
- The successor form delivers the unchecked items: live resource + pricing summary
  (`form/UniverseForm.tsx:196` mounts `UniverseResourceContainer`; `sections/universe-resource/
  UniverseResource.tsx:74-106` renders cores/memory/storage and `YBCost` per day/month from
  `pricePerHour`; `fields/PlacementsField/PlacementsFieldHelper.ts:283` calls `api.universeResource`
  on every configure; `utils/api.ts:140-141` POSTs `/universe_resources`; server route
  `managed/src/main/resources/v1.routes:257`) -- the #6379 objection is moot because this form does
  issue interim `/universe_configure` calls; full-move confirmation (`action-modals/FullMoveModal.tsx`,
  wired at `EditUniverse.tsx:72,257`; en.json "Confirm Full Move Update"); auth toggles
  (`sections/security/SecurityConfiguration.tsx:151,156` `<YSQLField>`/`<YCQLField>` backed by
  `UserIntent.enableYSQLAuth`/`enableYCQLAuth`, `UniverseDefinitionTaskParams.java:674-678`; #6440 closed
  2021-07-27). Sections rendered at `form/UniverseForm.tsx:287-303`: Cloud, Instance, Security, Advanced,
  GFlags, HelmOverrides, UserTags.
- The features-v2 wizard (`managed/ui/src/redesign/features-v2/universe/create-universe/`, route
  `/create-universe` `routes.jsx:254`, gated by `isUniverseRevampExperienceEnabled`
  `UniverseView.jsx:596`) reinstates a discrete review step (`CreateUniverseContext.ts:43-53`
  `REVIEW = 9`) with a live cost estimate (`steps/review-summary/ReviewAndSummary.tsx`:
  `getUniverseResources(payload)` -> `price_per_hour` daily/monthly) -- the #6439 ask.
- Residual (#6438): no route-leave/unsaved-changes guard exists in either form (0 files matching
  `beforeunload|routerWillLeave|setRouteLeaveHook|useBlocker|<Prompt` under `universe-form/` or
  `features-v2/universe/create-universe/` on every target); the V2 Cancel/Close handlers are
  `window.location.href = '/'` (master HEAD 52c3c2bf58 `CreateUniverse.tsx`; no handler at all on
  v2025.2.6.0); the 2023 form's "Reset Config" button resets placements immediately
  (`PlacementsField.tsx:242` `onClick={() => setValue(RESET_AZ_FIELD, true)}`).

Still affected: no required target for the tracker's substance. The two #6438 dialogs are absent on all
five targets; they are cosmetic UX items already tracked by open issue #6438 and do not warrant a master
fix PR under #4578.

## Trigger matrix
Predicted from source before the per-tag run (2023 form on all five; V2 wizard on 2025.2/2026.1/master
HEAD only; #6438 leave/reset dialogs absent everywhere). Observed = `yb4578-verify-ui.txt`.

| condition | expected | observed (2024.2 / 2025.1 / 2025.2 / 2026.1 / master) |
|---|---|---|
| 2020 #4578 wizard (`redesign/universe/wizard/`) present | absent on all (removed v2.17.3.0) | absent x5 (also 404 at master HEAD 52c3c2bf58) |
| 2023 form default: `UniverseForm.tsx` renderSections + `enableNewUI: true` + `/universes/create` -> UniverseNewView | present x5 | PASS x5 (renderSections :287; enableNewUI :29/:28/:28/:27/:28; routes :261/:261/:273/:280/:261) |
| Cloud/Instance/Security/Advanced/GFlags/UserTags sections | present x5 | PASS x5 (:290/:291/:294/:295/:298/:300) |
| #6437/#6379 live pricing summary (`UniverseResource.tsx` pricePerHour; `api.universeResource` on configure; POST `/universe_resources`) | present x5 | PASS x5 (:74; helper :280/:283; api :129/:141; v1.routes :256/:257) |
| #6438 full-move confirmation (`FullMoveModal.tsx`, `showFMModal`) | present x5 | PASS x5 (:65/:72) |
| #6440 auth toggles (`<YSQLField>`, `enableYSQLAuth`) | present x5 | PASS x5 (:140/:139/:150/:151/:139; java :478/:484/:590/:630/:480) |
| #6439 discrete review step with cost (V2 `REVIEW` + `price_per_hour`) | absent 2024.2, 2025.1, 2.25.2.0 preview; present 2025.2, 2026.1, master HEAD | matches: absent / absent / PASS (:137) / PASS (:153) / preview absent, HEAD PASS |
| #6438 leave-unsaved-form confirmation | absent x5 | GAP x5 (0 matching files; V2 Cancel no handler on 2025.2, `href='/'` on 2026.1/master) |
| #6438 Reset Config confirmation | absent x5 | GAP x5 (`resetAZLabel` button present :427/:448/:500/:799/:437, no modal) |
| Control: DB image contains YBA/yugaware or universe-form assets | NO_YBA x5 | NO_YBA x5; digest OK x5; in-image revision == tag SHA x5 |

## Reproduction
No executable reproduction exists with the official `yugabytedb/yugabyte` images: the subject is a YBA UI
form and YBA ships only as the separate `yugabytedb/yugaware` image (plus PostgreSQL/Prometheus). A live
YBA deployment is out of scope. The matrix is grounded in (a) read-only source inspection at the exact
resolved tag SHAs, (b) GitHub API inspection at the controller's master HEAD, and (c) an observed image
control per resolved DB image. Files (all under `/tmp/claude/yb4578/`):

```sh
bash /tmp/claude/yb4578/yb4578-verify-ui-sh.txt   > /tmp/claude/yb4578/yb4578-verify-ui.txt     # per-tag source checks (git object reads)
bash /tmp/claude/yb4578/yb4578-image-check-sh.txt > /tmp/claude/yb4578/yb4578-image-check.txt   # podman control, 5 images, NO_YBA
python3 /tmp/claude/yb4578/yb4578-dockerhub-query-py.txt > /tmp/claude/yb4578/yb4578-dockerhub-freshness.txt
# master HEAD evidence (GitHub API, mcpproxy github:get_file_contents): /tmp/claude/yb4578/yb4578-master-head-check.txt
# consolidated table: /tmp/claude/yb4578/yb4578-matrix.txt
```

Failure signature that would indicate the reported gap: `2020_WIZARD present` or any `[PASS]` line
reporting `ABSENT` for the 2023-form checks in `yb4578-verify-ui.txt`, or `YBA_FILES_FOUND` in the image
control (which would make a runtime UI repro possible and required). Observed: none. The `[GAP]` lines are
the expected #6438 residuals (predicted before the run). For a manual check against a live YBA: open
`/universes/create` (2023 form: sections, cost-estimator widget `data-testid="cost-estimator"`) and, with
the universe-revamp runtime flag, `/create-universe` (V2 wizard, step 9 "Review & Summary" with daily and
monthly cost).

## Versions tested
Resolved at: 2026-09-05T21:28Z (controller; Docker Hub tag metadata + `github:get_commit`).
Resolution evidence: independently re-queried 2026-09-05T21:45:54Z
(`/tmp/claude/yb4578/yb4578-dockerhub-freshness.txt`): Docker Hub
`https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=<prefix>&ordering=last_updated`
returns newest 2024.2 = 2024.2.11.0-b36 (pushed 2026-08-25T16:02:12Z; prior 2024.2.10.0-b62), 2025.1 =
2025.1.4.0-b103 (2026-04-20T15:59:42Z; prior 2025.1.3.2-b1), 2025.2 = 2025.2.6.0-b111 (2026-09-04T22:23:02Z;
prior 2025.2.5.2-b5), 2026.1 = 2026.1.1.1-b2 (2026-08-18T20:03:16Z; prior 2026.1.1.0-b91; `latest` has the
same arm64 digest sha256:d27eede7...), 2.25. = 2.25.2.0-b359 (2025-05-20T18:34:42Z); `name=2.27.`,
`name=2.29.`, `name=2026.2` return count 0 (no newer preview line). Tag -> commit verified locally
(`git rev-parse <tag>^{commit}`) and in-image (`yb-master --version` revision) for all five; master HEAD
52c3c2bf5800aad3e9ae74f4d41b224868caac09 (2026-09-05T19:58:56Z) confirmed via `github:list_commits`.

aarch64 image digests: 2024.2.11.0-b36 sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288;
2025.1.4.0-b103 sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0; 2025.2.6.0-b111
sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e; 2026.1.1.1-b2
sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699 (multi-arch manifest
sha256:3926eedf0ff4b99426fbff533128ff16635ed5697a5cf086dde4fafec28e6f75); 2.25.2.0-b359
sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5.

| target | latest resolved version | source revision | observed repro/control | effective status |
|---|---|---|---|---|
| 2024.2 | 2024.2.11.0-b36 (tag v2024.2.11.0; image docker.io/yugabytedb/yugabyte:2024.2.11.0-b36-aarch64) | 04544fd94fb4e50d05994784f444b075a4a5dfdb | Runtime repro not applicable: DB image `NO_YBA`, digest OK, in-image revision == tag SHA. Source at tag: 2020 wizard absent; 2023 form default (`UniverseForm.tsx:287-300`, `feature.js:29`, `routes.js:261`); pricing summary `UniverseResource.tsx:74`, `PlacementsFieldHelper.ts:280`, `api.ts:129`, `v1.routes:256`; `FullMoveModal.tsx` + `EditUniverse.tsx:65`; `SecurityConfiguration.tsx:140`; no V2 wizard; #6438 leave/reset dialogs absent. | fixed (redesign superseded and shipped; #6438 leave/reset dialogs outstanding, cosmetic) |
| 2025.1 | 2025.1.4.0-b103 (tag v2025.1.4.0; ...:2025.1.4.0-b103-aarch64) | 690ac4141e57d060fd834312f396643e8c41f260 | Not applicable (`NO_YBA`, digest OK, revision == tag SHA). Source: 2020 wizard absent; 2023 form default (`feature.js:28`, `routes.js:261`); pricing `UniverseResource.tsx:74`, helper `:283`, `api.ts:141`, `v1.routes:257`; full-move `EditUniverse.tsx:72`; auth `SecurityConfiguration.tsx:139`; no V2 wizard; #6438 leave/reset absent. | fixed (same residual) |
| 2025.2 | 2025.2.6.0-b111 (tag v2025.2.6.0; ...:2025.2.6.0-b111-aarch64) | ae22ef35ff65de63303467d95ac3e6d2382feba9 | Not applicable (`NO_YBA`, digest OK, revision == tag SHA). Source: 2020 wizard absent; 2023 form default (`feature.js:28`, `routes.js:273`); pricing/full-move/auth PASS (`SecurityConfiguration.tsx:150`); V2 wizard present (`routes.js:251`, `REVIEW` enum `CreateUniverseContext.ts:42`, `ReviewAndSummary.tsx:137` `price_per_hour`); V2 Cancel button has no handler; #6438 leave/reset absent. | fixed (same residual) |
| 2026.1 | 2026.1.1.1-b2 (tag v2026.1.1.1; = `latest`; ...:2026.1.1.1-b2-aarch64) | 3b010e021f6027744bf9e016b17e108a2eb24e44 | Not applicable (`NO_YBA`, digest OK, revision == tag SHA). Source: 2020 wizard absent; 2023 form default (`feature.js:27`, `routes.jsx:280`); pricing/full-move/auth PASS (`SecurityConfiguration.tsx:151`); V2 wizard present (`routes.jsx:252`, `CreateUniverseContext.ts:43`, `ReviewAndSummary.tsx:153`); V2 Cancel `window.location.href = '/'` (`CreateUniverse.tsx:82`) with no confirmation; #6438 leave/reset absent. | fixed (same residual) |
| master | preview 2.25.2.0-b359 (tag v2.25.2.0; ...:2.25.2.0-b359-aarch64); master HEAD 52c3c2bf58 (2026-09-05T19:58:56Z); local checkout 46a9f7adfd | 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d (HEAD 52c3c2bf5800aad3e9ae74f4d41b224868caac09) | Preview image: not applicable (`NO_YBA`, digest OK, revision == tag SHA); source at v2.25.2.0: 2020 wizard absent, 2023 form default (`feature.js:28`, `routes.js:261`), pricing/full-move/auth PASS, no V2 wizard (predates PLAT-16231). Master HEAD via GitHub API: `redesign/universe/` 404; `feature.js`, `UniverseResource.tsx`, `FullMoveModal.tsx` blobs identical to local HEAD; V2 `ReviewAndSummary.tsx` (blob f1777c8f) computes daily/monthly cost from `getUniverseResources`; V2 `CreateUniverse.tsx` (blob 4596ba28) Cancel/Close -> `window.location.href = '/'`; #6438 leave/reset absent. | fixed (same residual); no master fix PR warranted under #4578 |

Reported release: none (2020 tracker, milestone 2.5.x; predates the current versioning). The #4578 wizard
itself shipped as a non-default "Try New UI" route from v2.3.0.0 (part 1) / v2.5.1 (last #4578 commit)
through v2.17.2.0 -- verified present at v2.5.0, v2.5.3, v2.6.0, v2.8.0, v2.14.0.0, v2.16.0.0, v2.17.2.0 and
absent from v2.17.3.0 onward. Not a current target; not separately runnable (no YBA images for those lines
are in scope).

First released version containing each closure (tag/commit ancestry via `git tag --contains`; provenance
only -- effectiveness is the per-target source verification above):

- #4578 wizard: 828927b095 (2020-07-20, part 1) -> v2.3.0.0; 7e45cb5f42 (2020-10-19, beta + "Try New UI"
  link) -> v2.3.3.0 / v2.4.0; c1fda7ffe3 (2020-11-23, last commit referencing #4578) -> v2.5.1.
- Redesigned form: ac66da2c15 PLAT-7179 (2023-01-30) -> v2.17.2.0 (behind `enableNewUI: false`);
  df6efc876f PLAT-7545 (2023-03-01, default true) -> v2.17.3.0, stable v2.18.0.0; therefore present on
  every 2.18+/2.20+/2024.x/2025.x/2026.x tag including all five targets.
- Wizard removal: f185c02e87 PLAT-7712 (2023-03-14) -> v2.17.3.0, stable v2.18.0.0.
- features-v2 wizard: 4767d3f3cc PLAT-16231 (2025-04-22) and 41481e8bfa PLAT-16236 review step
  (2025-07-03) -> v2025.2.0.0 (hence 2025.2.6.0, 2026.1.1.1, master HEAD; absent from 2024.2, 2025.1 and
  the 2.25.2.0 preview).
- #6440 `enableYSQLAuth`/`enableYCQLAuth`: closed 2021-07-27; present on all five targets.
- #6379, #6437, #6438, #6439: still OPEN on GitHub (checked 2026-09-05); no commit references them
  in the reachable history. Their intent is delivered by the successors as described, except the two
  #6438 dialogs.

## Proposed fix direction
No code change under #4578. Recommend closing #4578 as completed/superseded: the tracked redesign shipped
(2020-2023) and was replaced by the default PLAT-7179 form and the features-v2 wizard; #6440 is closed;
#6379/#6437/#6439 are satisfied by the successors and can be closed as obsolete (the `/universe_resources`
+ `/universe_configure` flow in `PlacementsFieldHelper.ts:283` removes #6379's premise). Keep #6438 open
(or re-scope it) for the two missing confirmations if the product still wants them: a route-leave guard
for dirty forms in `redesign/features/universe/universe-form/form/UniverseForm.tsx` and
`features-v2/universe/create-universe/CreateUniverse.tsx` (Cancel/Close handlers), and a confirmation
before `setValue(RESET_AZ_FIELD, true)` at `PlacementsField.tsx:242`. Alternatives considered: treating
the missing dialogs as blocking for #4578 -- rejected because they are UX polish items with their own
tracker and the tracker's substance (the redesigned form) is fully present.

## Test strategy
No fix under #4578, so no new test. If the #6438 dialogs are implemented, cover them with the existing
`managed/ui` Jest setup (`npm run test:ci` in `managed/ui/`) via component tests next to
`redesign/features/universe/universe-form/` (e.g. a test asserting a confirm dialog before
`RESET_AZ_FIELD` is set, and a route-leave prompt when the form is dirty), exercising public UI entry
points only (`/universes/create`, `/create-universe`). Existing V2 utilities tests live at
`features-v2/universe/create-universe/CreateUniverseUtils.*.test.ts`.

## Confidentiality check
Contents are public: GitHub issue/sub-issue text and states, public commit SHAs and tags, public Phorge
revision IDs from commit messages, official Docker Hub tags/digests, and source paths/line numbers from
the public repository. Commit author e-mail addresses and the internal test-plan hostnames in the 2020
commit messages were not copied into this artifact or the evidence files. No customer identifiers, PII,
internal hostnames/URLs, or secrets.

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-verify-ui-sh.txt`):

#!/usr/bin/env bash
# yugabyte/yugabyte-db#4578 -- per-tag source verification of the YBA Create/Edit universe form.
# Read-only against the local yugabyte-db checkout (git object inspection only; no working-tree changes).
# For each required target it records: (1) the local tag SHA equals the controller-resolved SHA,
# (2) the 2020 "#4578 wizard" (redesign/universe/wizard) is absent, (3) the 2023 redesigned form
# (redesign/features/universe/universe-form) is present, routed at /universes/create and enabled by
# default, (4) the delivered pieces behind the unchecked sub-steps (#6379/#6437 live pricing summary,
# #6438 full-move confirmation, #6439 review step, #6440 auth toggles), (5) the residual gaps
# (#6438 leave-unsaved-form and Reset-Config confirmations), (6) the features-v2 wizard presence.
# Usage: bash /tmp/claude/yb4578/yb4578-verify-ui-sh.txt > /tmp/claude/yb4578/yb4578-verify-ui.txt
set -u
REPO=/Users/brandonmcnama/src/github.com/yugabyte/yugabyte-db
cd "$REPO" || exit 1
TARGETS=(
  "2024.2|v2024.2.11.0|04544fd94fb4e50d05994784f444b075a4a5dfdb|routes.js"
  "2025.1|v2025.1.4.0|690ac4141e57d060fd834312f396643e8c41f260|routes.js"
  "2025.2|v2025.2.6.0|ae22ef35ff65de63303467d95ac3e6d2382feba9|routes.js"
  "2026.1|v2026.1.1.1|3b010e021f6027744bf9e016b17e108a2eb24e44|routes.jsx"
  "master-preview|v2.25.2.0|03bbc9f31eae5c3fecabd8f0afa5e99211e6080d|routes.js"
)
UI=managed/ui/src
F2020=$UI/redesign/universe/wizard/UniverseWizard.tsx
F2020B=$UI/redesign/universes/wizard/UniverseWizard.jsx
FORM=$UI/redesign/features/universe/universe-form
V2=$UI/redesign/features-v2/universe/create-universe
V2OLD=$UI/redesign/features-v2/universe-form-wizard

exists() { git cat-file -e "$1:$2" 2>/dev/null; }
has() { git show "$1:$2" 2>/dev/null | rg -q -- "$3"; }
line() { git show "$1:$2" 2>/dev/null | rg -n -- "$3" | head -1 | cut -d: -f1; }
chk() { # chk <label> <expect PASS|GAP> <cond exit code>
  local label=$1 expect=$2 rc=$3
  if [ "$rc" -eq 0 ]; then echo "  [$expect] $label -> present"; else echo "  [$expect] $label -> ABSENT"; fi
}

echo "# yb4578 per-tag source verification; run at $(python3 -c 'import datetime;print(datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ"))')"
echo "# repo=$REPO local HEAD=$(git rev-parse HEAD)"
for e in "${TARGETS[@]}"; do
  IFS='|' read -r target tag want routes <<<"$e"
  got=$(git rev-parse "$tag^{commit}" 2>/dev/null)
  echo
  echo "== $target $tag"
  if [ "$got" = "$want" ]; then echo "  SHA OK $got"; else echo "  SHA MISMATCH got=$got want=$want"; fi
  echo "  tag commit: $(git log -1 --format='%cI %s' "$tag^{commit}" | cut -c1-120)"

  # (2) 2020 wizard removed (PLAT-7712 f185c02e87)
  if exists "$tag" "$F2020" || exists "$tag" "$F2020B"; then echo "  2020_WIZARD present (unexpected)"; else echo "  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)"; fi

  # (3) 2023 redesigned form present, default, routed
  chk "2023 form $FORM/form/UniverseForm.tsx renderSections (line $(line "$tag" "$FORM/form/UniverseForm.tsx" 'renderSections = '))" PASS $(has "$tag" "$FORM/form/UniverseForm.tsx" 'renderSections'; echo $?)
  for s in CloudConfiguration InstanceConfiguration SecurityConfiguration AdvancedConfiguration GFlags UserTags; do
    chk "  section <$s> rendered (UniverseForm.tsx line $(line "$tag" "$FORM/form/UniverseForm.tsx" "<$s"))" PASS $(has "$tag" "$FORM/form/UniverseForm.tsx" "<$s"; echo $?)
  done
  chk "feature.js enableNewUI: true (line $(line "$tag" "$UI/reducers/feature.js" 'enableNewUI: true'))" PASS $(has "$tag" "$UI/reducers/feature.js" 'enableNewUI: true'; echo $?)
  chk "$routes /universes/create -> UniverseNewView (line $(line "$tag" "$UI/$routes" 'path="/universes/create" component=\{UniverseNewView\}'))" PASS $(has "$tag" "$UI/$routes" 'path="/universes/create" component=\{UniverseNewView\}'; echo $?)
  chk "UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js)" PASS $( { has "$tag" "$UI/pages/UniverseNewView.jsx" 'UniverseFormContainer' || has "$tag" "$UI/pages/UniverseNewView.js" 'UniverseFormContainer'; } ; echo $?)

  # (4) sub-step deliverables
  chk "#6437 summary widget with pricing: $FORM/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line $(line "$tag" "$FORM/form/sections/universe-resource/UniverseResource.tsx" 'pricePerHour'))" PASS $(has "$tag" "$FORM/form/sections/universe-resource/UniverseResource.tsx" 'pricePerHour'; echo $?)
  chk "#6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line $(line "$tag" "$FORM/form/UniverseForm.tsx" '<UniverseResourceContainer'))" PASS $(has "$tag" "$FORM/form/UniverseForm.tsx" '<UniverseResourceContainer'; echo $?)
  chk "#6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line $(line "$tag" "$FORM/form/fields/PlacementsField/PlacementsFieldHelper.ts" 'api.universeResource'))" PASS $(has "$tag" "$FORM/form/fields/PlacementsField/PlacementsFieldHelper.ts" 'api.universeResource'; echo $?)
  chk "#6379 endpoint: utils/api.ts /universe_resources (line $(line "$tag" "$FORM/utils/api.ts" 'universe_resources'))" PASS $(has "$tag" "$FORM/utils/api.ts" 'universe_resources'; echo $?)
  chk "#6379 server route v1.routes POST universe_resources (line $(line "$tag" "managed/src/main/resources/v1.routes" 'POST.*/universe_resources'))" PASS $(has "$tag" "managed/src/main/resources/v1.routes" 'POST.*/universe_resources'; echo $?)
  chk "#6438 full-move confirmation: $FORM/action-modals/FullMoveModal.tsx" PASS $(exists "$tag" "$FORM/action-modals/FullMoveModal.tsx"; echo $?)
  chk "#6438 full-move wired: EditUniverse.tsx showFMModal (line $(line "$tag" "$FORM/EditUniverse.tsx" 'showFMModal'))" PASS $(has "$tag" "$FORM/EditUniverse.tsx" 'showFMModal'; echo $?)
  chk "#6438 full-move text: en.json 'Confirm Full Move Update' (line $(line "$tag" "$UI/translations/en.json" 'Confirm Full Move Update'))" PASS $(has "$tag" "$UI/translations/en.json" 'Confirm Full Move Update'; echo $?)
  chk "#6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line $(line "$tag" "$FORM/form/sections/security/SecurityConfiguration.tsx" '<YSQLField'))" PASS $(has "$tag" "$FORM/form/sections/security/SecurityConfiguration.tsx" '<YSQLField'; echo $?)
  chk "#6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line $(line "$tag" "managed/src/main/java/com/yugabyte/yw/forms/UniverseDefinitionTaskParams.java" 'enableYSQLAuth'))" PASS $(has "$tag" "managed/src/main/java/com/yugabyte/yw/forms/UniverseDefinitionTaskParams.java" 'enableYSQLAuth'; echo $?)

  # (5) residual gaps (#6438 remaining modals). GAP = expected absent.
  leave=$(git grep -l -E 'beforeunload|routerWillLeave|setRouteLeaveHook|useBlocker|<Prompt' "$tag" -- "$FORM" "$V2" "$V2OLD" 2>/dev/null | wc -l | tr -d ' ')
  echo "  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: $leave (0 = not implemented)"
  chk "#6438 Reset Config button (en.json resetAZLabel 'Reset Config', line $(line "$tag" "$UI/translations/en.json" '"resetAZLabel": "Reset Config"'))" PASS $(has "$tag" "$UI/translations/en.json" '"resetAZLabel": "Reset Config"'; echo $?)
  reset_modal=$(git grep -l -i -E 'resetAZ.*(Modal|confirm)|confirm.*resetAZ' "$tag" -- "$FORM" 2>/dev/null | wc -l | tr -d ' ')
  echo "  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: $reset_modal (0 = not implemented)"

  # (6) features-v2 wizard (2025+), #6439-style discrete Review step
  if exists "$tag" "$V2/CreateUniverse.tsx"; then
    echo "  V2_WIZARD present: $V2/CreateUniverse.tsx; route $(git show "$tag:$UI/$routes" | rg -n 'path="/create-universe"' | head -1 | cut -d: -f1 | sed 's/^/'"$routes"':/')"
    chk "  V2 review step with cost: steps/review-summary/ReviewAndSummary.tsx price_per_hour (line $(line "$tag" "$V2/steps/review-summary/ReviewAndSummary.tsx" 'price_per_hour'))" PASS $(has "$tag" "$V2/steps/review-summary/ReviewAndSummary.tsx" 'price_per_hour'; echo $?)
    chk "  V2 steps enum REVIEW (CreateUniverseContext.ts line $(line "$tag" "$V2/CreateUniverseContext.ts" 'REVIEW = '))" PASS $(has "$tag" "$V2/CreateUniverseContext.ts" 'REVIEW = '; echo $?)
    echo "  [GAP] V2 cancel button navigates without confirmation: CreateUniverse.tsx line $(line "$tag" "$V2/CreateUniverse.tsx" "window.location.href = '/'") (window.location.href = '/')"
  elif exists "$tag" "$V2OLD/CreateUniverse.tsx"; then
    echo "  V2_WIZARD present at pre-rename path $V2OLD"
  else
    echo "  V2_WIZARD absent (features-v2 create-universe not on this tag)"
  fi
done
echo
echo "# provenance (tag ancestry, not runtime proof)"
first_v() { git tag --contains "$1" 2>/dev/null | grep -E '^v[0-9]' | sort -V | head -"${2:-3}" | tr '\n' ' '; }
echo "828927b095 2020-07-20 [YW][#4578 - part 1] first v-tags: $(first_v 828927b095 3)"
echo "7e45cb5f42 2020-10-19 [YW][#4578] Create universe redesign (beta) first v-tags: $(first_v 7e45cb5f42 3)"
echo "c1fda7ffe3 2020-11-23 [YW][#4578] Fixes and improvements (last #4578 commit) first v-tags: $(first_v c1fda7ffe3 3)"
echo "ac66da2c15 2023-01-30 [PLAT-7179] universe_code_revamp merge first v-tags: $(first_v ac66da2c15 3)"
echo "df6efc876f 2023-03-01 [PLAT-7545] enableNewUI default true first v-tags: $(first_v df6efc876f 3)"
echo "f185c02e87 2023-03-14 [PLAT-7712] Remove wizard flow first v-tags: $(first_v f185c02e87 3)"
echo "4767d3f3cc 2025-04-22 [PLAT-16231] features-v2 create universe folder first v-tags: $(first_v 4767d3f3cc 3)"
echo "41481e8bfa 2025-07-03 [PLAT-16236] V2 review-summary first v-tags: $(first_v 41481e8bfa 3)"
echo "2020 wizard present at: $(for t in v2.5.0 v2.5.3 v2.6.0 v2.8.0 v2.14.0.0 v2.16.0.0 v2.17.2.0; do exists $t $F2020 && printf '%s ' $t; done)"
echo "2020 wizard absent at: $(for t in v2.17.3.0 v2.18.0.0 v2.20.0.0 v2024.2.11.0 v2025.1.4.0 v2025.2.6.0 v2026.1.1.1 v2.25.2.0; do exists $t $F2020 || printf '%s ' $t; done)"


[PLAT-7179]: https://yugabyte.atlassian.net/browse/PLAT-7179?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-image-check-sh.txt`):

#!/usr/bin/env bash
# yugabyte/yugabyte-db#4578 -- runtime-applicability control for the five resolved
# docker.io/yugabytedb/yugabyte images. #4578 is a YugabyteDB Anywhere (managed/ui) form
# redesign; YBA ships only as the separate yugabytedb/yugaware image, so this control records,
# per DB image: (1) pulled digest == controller-resolved aarch64 digest, (2) in-image
# `yb-master --version` revision == the verified source tag SHA, (3) DB daemons present,
# (4) no YBA/yugaware component or UI bundle (yugaware jar/dir, yb-platform-*, yba-installer,
# com/yugabyte/yw classes, universe-form JS chunks). Result "NO_YBA" per image = the executable
# UI reproduction is not applicable to that image, so the matrix cell is grounded in source.
# Usage: bash /tmp/claude/yb4578/yb4578-image-check-sh.txt > /tmp/claude/yb4578/yb4578-image-check.txt
set -u
IMAGES=(
  "2024.2|docker.io/yugabytedb/yugabyte:2024.2.11.0-b36-aarch64|sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288|04544fd94fb4e50d05994784f444b075a4a5dfdb"
  "2025.1|docker.io/yugabytedb/yugabyte:2025.1.4.0-b103-aarch64|sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0|690ac4141e57d060fd834312f396643e8c41f260"
  "2025.2|docker.io/yugabytedb/yugabyte:2025.2.6.0-b111-aarch64|sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e|ae22ef35ff65de63303467d95ac3e6d2382feba9"
  "2026.1|docker.io/yugabytedb/yugabyte:2026.1.1.1-b2-aarch64|sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699|3b010e021f6027744bf9e016b17e108a2eb24e44"
  "master|docker.io/yugabytedb/yugabyte:2.25.2.0-b359-aarch64|sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5|03bbc9f31eae5c3fecabd8f0afa5e99211e6080d"
)
echo "# yb4578 image control; run at $(python3 -c 'import datetime;print(datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ"))') via podman $(podman --version 2>/dev/null | awk '{print $3}') (aarch64 host)"
for e in "${IMAGES[@]}"; do
  IFS='|' read -r target img want_digest want_rev <<<"$e"
  echo "== $target $img =="
  got=$(podman image inspect "$img" --format '{{index .RepoDigests 0}}' 2>/dev/null | sed 's/.*@//')
  if [ "$got" = "$want_digest" ]; then echo "$target DIGEST OK $got"; else echo "$target DIGEST MISMATCH got=$got want=$want_digest"; fi
  podman run --rm --name "yb4578-imgcheck-$target" --entrypoint /bin/bash "$img" -c '
    ver=$(/home/yugabyte/bin/yb-master --version 2>/dev/null | head -1)
    echo "yb-master --version: ${ver:-<none>}"
    rev=$(echo "$ver" | sed -n "s/.*revision \([0-9a-f]*\).*/\1/p")
    if [ "$rev" = "'"$want_rev"'" ]; then echo "REVISION OK $rev == tag SHA"; else echo "REVISION MISMATCH got=$rev want='"$want_rev"'"; fi
    for b in yb-master yb-tserver yugabyted ysqlsh; do
      [ -x /home/yugabyte/bin/$b ] && echo "DB_BINARY_PRESENT /home/yugabyte/bin/$b"
    done
    # YBA ships as the Play app "yugaware" (jar/dir, /opt/yugabyte/yugaware, yb-platform-*), never in DB images.
    hits=$(find / -xdev \( -iname "*yugaware*" -o -iname "yb-platform*" -o -iname "yba-installer*" -o -ipath "*/com/yugabyte/yw/*" -o -iname "UniverseForm*" -o -ipath "*/universe-form/*" -o -ipath "*/create-universe/*" \) -not -path "/proc/*" 2>/dev/null | head -5)
    if [ -z "$hits" ]; then echo "NO_YBA: no yugaware/YBA/universe-form files in image"; else echo "YBA_FILES_FOUND:"; echo "$hits"; fi
    # the only web assets in a DB image are the yb-master/yb-tserver admin pages under /home/yugabyte/www
    echo "www assets: $(ls /home/yugabyte/www 2>/dev/null | tr "\n" " ")"
    echo "home: $(ls /home/yugabyte 2>/dev/null | tr "\n" " ")"
  ' 2>&1 | sed "s/^/$target /"
done

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-verify-ui.txt`):

# yb4578 per-tag source verification; run at 2026-09-05T21:41:08Z
# repo=/Users/brandonmcnama/src/github.com/yugabyte/yugabyte-db local HEAD=46a9f7adfd226d7bc3315c38b660eef69bb11196

== 2024.2 v2024.2.11.0
  SHA OK 04544fd94fb4e50d05994784f444b075a4a5dfdb
  tag commit: 2026-08-17T14:41:57-07:00 [BACKPORT 2024.2][DEVOPS-3852] build: pull request workflow triggers (#33242) (#33329)
  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)
  [PASS] 2023 form managed/ui/src/redesign/features/universe/universe-form/form/UniverseForm.tsx renderSections (line 287) -> present
  [PASS]   section <CloudConfiguration> rendered (UniverseForm.tsx line 290) -> present
  [PASS]   section <InstanceConfiguration> rendered (UniverseForm.tsx line 291) -> present
  [PASS]   section <SecurityConfiguration> rendered (UniverseForm.tsx line 294) -> present
  [PASS]   section <AdvancedConfiguration> rendered (UniverseForm.tsx line 295) -> present
  [PASS]   section <GFlags> rendered (UniverseForm.tsx line 298) -> present
  [PASS]   section <UserTags> rendered (UniverseForm.tsx line 300) -> present
  [PASS] feature.js enableNewUI: true (line 29) -> present
  [PASS] routes.js /universes/create -> UniverseNewView (line 261) -> present
  [PASS] UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js) -> present
  [PASS] #6437 summary widget with pricing: managed/ui/src/redesign/features/universe/universe-form/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line 74) -> present
  [PASS] #6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line 196) -> present
  [PASS] #6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line 280) -> present
  [PASS] #6379 endpoint: utils/api.ts /universe_resources (line 129) -> present
  [PASS] #6379 server route v1.routes POST universe_resources (line 256) -> present
  [PASS] #6438 full-move confirmation: managed/ui/src/redesign/features/universe/universe-form/action-modals/FullMoveModal.tsx -> present
  [PASS] #6438 full-move wired: EditUniverse.tsx showFMModal (line 65) -> present
  [PASS] #6438 full-move text: en.json 'Confirm Full Move Update' (line 640) -> present
  [PASS] #6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line 140) -> present
  [PASS] #6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line 478) -> present
  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: 0 (0 = not implemented)
  [PASS] #6438 Reset Config button (en.json resetAZLabel 'Reset Config', line 427) -> present
  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: 0 (0 = not implemented)
  V2_WIZARD absent (features-v2 create-universe not on this tag)

== 2025.1 v2025.1.4.0
  SHA OK 690ac4141e57d060fd834312f396643e8c41f260
  tag commit: 2026-04-02T16:48:13Z [BACKPORT 2025.1][#30772] DocDB: Fix "thread pool not ready error" for large txn applying
  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)
  [PASS] 2023 form managed/ui/src/redesign/features/universe/universe-form/form/UniverseForm.tsx renderSections (line 287) -> present
  [PASS]   section <CloudConfiguration> rendered (UniverseForm.tsx line 290) -> present
  [PASS]   section <InstanceConfiguration> rendered (UniverseForm.tsx line 291) -> present
  [PASS]   section <SecurityConfiguration> rendered (UniverseForm.tsx line 294) -> present
  [PASS]   section <AdvancedConfiguration> rendered (UniverseForm.tsx line 295) -> present
  [PASS]   section <GFlags> rendered (UniverseForm.tsx line 298) -> present
  [PASS]   section <UserTags> rendered (UniverseForm.tsx line 300) -> present
  [PASS] feature.js enableNewUI: true (line 28) -> present
  [PASS] routes.js /universes/create -> UniverseNewView (line 261) -> present
  [PASS] UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js) -> present
  [PASS] #6437 summary widget with pricing: managed/ui/src/redesign/features/universe/universe-form/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line 74) -> present
  [PASS] #6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line 196) -> present
  [PASS] #6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line 283) -> present
  [PASS] #6379 endpoint: utils/api.ts /universe_resources (line 141) -> present
  [PASS] #6379 server route v1.routes POST universe_resources (line 257) -> present
  [PASS] #6438 full-move confirmation: managed/ui/src/redesign/features/universe/universe-form/action-modals/FullMoveModal.tsx -> present
  [PASS] #6438 full-move wired: EditUniverse.tsx showFMModal (line 72) -> present
  [PASS] #6438 full-move text: en.json 'Confirm Full Move Update' (line 667) -> present
  [PASS] #6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line 139) -> present
  [PASS] #6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line 484) -> present
  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: 0 (0 = not implemented)
  [PASS] #6438 Reset Config button (en.json resetAZLabel 'Reset Config', line 448) -> present
  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: 0 (0 = not implemented)
  V2_WIZARD absent (features-v2 create-universe not on this tag)

== 2025.2 v2025.2.6.0
  SHA OK ae22ef35ff65de63303467d95ac3e6d2382feba9
  tag commit: 2026-09-02T08:30:03-07:00 [BACKPORT 2025.2.6][#33558] YSQL: evict empty streams from MergingPgDocOpFetchStream
  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)
  [PASS] 2023 form managed/ui/src/redesign/features/universe/universe-form/form/UniverseForm.tsx renderSections (line 287) -> present
  [PASS]   section <CloudConfiguration> rendered (UniverseForm.tsx line 290) -> present
  [PASS]   section <InstanceConfiguration> rendered (UniverseForm.tsx line 291) -> present
  [PASS]   section <SecurityConfiguration> rendered (UniverseForm.tsx line 294) -> present
  [PASS]   section <AdvancedConfiguration> rendered (UniverseForm.tsx line 295) -> present
  [PASS]   section <GFlags> rendered (UniverseForm.tsx line 298) -> present
  [PASS]   section <UserTags> rendered (UniverseForm.tsx line 300) -> present
  [PASS] feature.js enableNewUI: true (line 28) -> present
  [PASS] routes.js /universes/create -> UniverseNewView (line 273) -> present
  [PASS] UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js) -> present
  [PASS] #6437 summary widget with pricing: managed/ui/src/redesign/features/universe/universe-form/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line 74) -> present
  [PASS] #6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line 196) -> present
  [PASS] #6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line 283) -> present
  [PASS] #6379 endpoint: utils/api.ts /universe_resources (line 141) -> present
  [PASS] #6379 server route v1.routes POST universe_resources (line 257) -> present
  [PASS] #6438 full-move confirmation: managed/ui/src/redesign/features/universe/universe-form/action-modals/FullMoveModal.tsx -> present
  [PASS] #6438 full-move wired: EditUniverse.tsx showFMModal (line 72) -> present
  [PASS] #6438 full-move text: en.json 'Confirm Full Move Update' (line 742) -> present
  [PASS] #6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line 150) -> present
  [PASS] #6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line 590) -> present
  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: 0 (0 = not implemented)
  [PASS] #6438 Reset Config button (en.json resetAZLabel 'Reset Config', line 500) -> present
  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: 0 (0 = not implemented)
  V2_WIZARD present: managed/ui/src/redesign/features-v2/universe/create-universe/CreateUniverse.tsx; route routes.js:251
  [PASS]   V2 review step with cost: steps/review-summary/ReviewAndSummary.tsx price_per_hour (line 137) -> present
  [PASS]   V2 steps enum REVIEW (CreateUniverseContext.ts line 42) -> present
  [GAP] V2 cancel button navigates without confirmation: CreateUniverse.tsx line  (window.location.href = '/')

== 2026.1 v2026.1.1.1
  SHA OK 3b010e021f6027744bf9e016b17e108a2eb24e44
  tag commit: 2026-08-13T23:45:33Z [BACKPORT 2026.1.1][PLAT-21546] Update postgres to version 14.23
  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)
  [PASS] 2023 form managed/ui/src/redesign/features/universe/universe-form/form/UniverseForm.tsx renderSections (line 287) -> present
  [PASS]   section <CloudConfiguration> rendered (UniverseForm.tsx line 290) -> present
  [PASS]   section <InstanceConfiguration> rendered (UniverseForm.tsx line 291) -> present
  [PASS]   section <SecurityConfiguration> rendered (UniverseForm.tsx line 294) -> present
  [PASS]   section <AdvancedConfiguration> rendered (UniverseForm.tsx line 295) -> present
  [PASS]   section <GFlags> rendered (UniverseForm.tsx line 298) -> present
  [PASS]   section <UserTags> rendered (UniverseForm.tsx line 300) -> present
  [PASS] feature.js enableNewUI: true (line 27) -> present
  [PASS] routes.jsx /universes/create -> UniverseNewView (line 280) -> present
  [PASS] UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js) -> present
  [PASS] #6437 summary widget with pricing: managed/ui/src/redesign/features/universe/universe-form/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line 74) -> present
  [PASS] #6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line 196) -> present
  [PASS] #6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line 283) -> present
  [PASS] #6379 endpoint: utils/api.ts /universe_resources (line 141) -> present
  [PASS] #6379 server route v1.routes POST universe_resources (line 257) -> present
  [PASS] #6438 full-move confirmation: managed/ui/src/redesign/features/universe/universe-form/action-modals/FullMoveModal.tsx -> present
  [PASS] #6438 full-move wired: EditUniverse.tsx showFMModal (line 72) -> present
  [PASS] #6438 full-move text: en.json 'Confirm Full Move Update' (line 1052) -> present
  [PASS] #6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line 151) -> present
  [PASS] #6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line 630) -> present
  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: 0 (0 = not implemented)
  [PASS] #6438 Reset Config button (en.json resetAZLabel 'Reset Config', line 799) -> present
  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: 0 (0 = not implemented)
  V2_WIZARD present: managed/ui/src/redesign/features-v2/universe/create-universe/CreateUniverse.tsx; route routes.jsx:252
  [PASS]   V2 review step with cost: steps/review-summary/ReviewAndSummary.tsx price_per_hour (line 153) -> present
  [PASS]   V2 steps enum REVIEW (CreateUniverseContext.ts line 43) -> present
  [GAP] V2 cancel button navigates without confirmation: CreateUniverse.tsx line 82 (window.location.href = '/')

== master-preview v2.25.2.0
  SHA OK 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d
  tag commit: 2025-05-08T10:37:54-07:00 [BACKPORT 2.25.2][#27104] DocDB: Update thirdparty to v20250507172143
  2020_WIZARD absent (removed by f185c02e87 PLAT-7712)
  [PASS] 2023 form managed/ui/src/redesign/features/universe/universe-form/form/UniverseForm.tsx renderSections (line 287) -> present
  [PASS]   section <CloudConfiguration> rendered (UniverseForm.tsx line 290) -> present
  [PASS]   section <InstanceConfiguration> rendered (UniverseForm.tsx line 291) -> present
  [PASS]   section <SecurityConfiguration> rendered (UniverseForm.tsx line 294) -> present
  [PASS]   section <AdvancedConfiguration> rendered (UniverseForm.tsx line 295) -> present
  [PASS]   section <GFlags> rendered (UniverseForm.tsx line 298) -> present
  [PASS]   section <UserTags> rendered (UniverseForm.tsx line 300) -> present
  [PASS] feature.js enableNewUI: true (line 28) -> present
  [PASS] routes.js /universes/create -> UniverseNewView (line 261) -> present
  [PASS] UniverseNewView -> UniverseFormContainer (pages/UniverseNewView.jsx|js) -> present
  [PASS] #6437 summary widget with pricing: managed/ui/src/redesign/features/universe/universe-form/form/sections/universe-resource/UniverseResource.tsx pricePerHour (line 74) -> present
  [PASS] #6437 widget mounted: UniverseForm.tsx <UniverseResourceContainer (line 196) -> present
  [PASS] #6379 live updates: PlacementsFieldHelper.ts api.universeResource on configure (line 283) -> present
  [PASS] #6379 endpoint: utils/api.ts /universe_resources (line 141) -> present
  [PASS] #6379 server route v1.routes POST universe_resources (line 257) -> present
  [PASS] #6438 full-move confirmation: managed/ui/src/redesign/features/universe/universe-form/action-modals/FullMoveModal.tsx -> present
  [PASS] #6438 full-move wired: EditUniverse.tsx showFMModal (line 72) -> present
  [PASS] #6438 full-move text: en.json 'Confirm Full Move Update' (line 653) -> present
  [PASS] #6440 auth toggles: SecurityConfiguration.tsx <YSQLField (line 139) -> present
  [PASS] #6440 server field: UniverseDefinitionTaskParams.java enableYSQLAuth (line 480) -> present
  [GAP] #6438 leave-unsaved-form confirmation in universe-form / features-v2 create-universe -> files matching: 0 (0 = not implemented)
  [PASS] #6438 Reset Config button (en.json resetAZLabel 'Reset Config', line 437) -> present
  [GAP] #6438 Reset Config confirmation modal in universe-form -> files matching: 0 (0 = not implemented)
  V2_WIZARD absent (features-v2 create-universe not on this tag)

# provenance (tag ancestry, not runtime proof)
828927b095 2020-07-20 [YW][#4578 - part 1] first v-tags: v1.0.0-beta-yugabyted-ui v2.3.0.0 v2.3.1.0 
7e45cb5f42 2020-10-19 [YW][#4578] Create universe redesign (beta) first v-tags: v1.0.0-beta-yugabyted-ui v2.3.3.0 v2.4.0 
c1fda7ffe3 2020-11-23 [YW][#4578] Fixes and improvements (last #4578 commit) first v-tags: v1.0.0-beta-yugabyted-ui v2.5.1 v2.5.2 
ac66da2c15 2023-01-30 [PLAT-7179] universe_code_revamp merge first v-tags: v2.17.2.0 v2.17.3.0 v2.18.0.0 
df6efc876f 2023-03-01 [PLAT-7545] enableNewUI default true first v-tags: v2.17.3.0 v2.18.0.0 v2.18.0.1 
f185c02e87 2023-03-14 [PLAT-7712] Remove wizard flow first v-tags: v2.17.3.0 v2.18.0.0 v2.18.0.1 
4767d3f3cc 2025-04-22 [PLAT-16231] features-v2 create universe folder first v-tags: v2025.2.0.0 v2025.2.0.1 v2025.2.1.0 
41481e8bfa 2025-07-03 [PLAT-16236] V2 review-summary first v-tags: v2025.2.0.0 v2025.2.0.1 v2025.2.1.0 
2020 wizard present at: v2.5.0 v2.5.3 v2.6.0 v2.8.0 v2.14.0.0 v2.16.0.0 v2.17.2.0 
2020 wizard absent at: v2.17.3.0 v2.18.0.0 v2.20.0.0 v2024.2.11.0 v2025.1.4.0 v2025.2.6.0 v2026.1.1.1 v2.25.2.0 


[DEVOPS-3852]: https://yugabyte.atlassian.net/browse/DEVOPS-3852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLAT-21546]: https://yugabyte.atlassian.net/browse/PLAT-21546?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-matrix.txt`):

# yugabyte/yugabyte-db#4578 "[Platform] Create/Edit universe form redesign" -- support-matrix summary
# Compiled 2026-09-05T21:47Z from:
#   /tmp/claude/yb4578/yb4578-verify-ui.txt        (per-tag source verification, run 2026-09-05T21:41:08Z)
#   /tmp/claude/yb4578/yb4578-master-head-check.txt (controller master HEAD 52c3c2bf58 via GitHub API, 2026-09-05T21:44Z)
#   /tmp/claude/yb4578/yb4578-image-check.txt       (podman control on the five DB images, run 2026-09-05T21:44:37Z)
#   /tmp/claude/yb4578/yb4578-dockerhub-freshness.txt (Docker Hub re-query 2026-09-05T21:45:54Z)
# Runtime reproduction of a YBA UI form is not possible with docker.io/yugabytedb/yugabyte images (they ship no
# YBA/yugaware component: NO_YBA x5). Each row is therefore grounded in source at the exact resolved tag SHA plus
# the image control (digest OK, in-image yb-master revision == tag SHA, NO_YBA).

Legend: W2020 = the #4578 wizard (managed/ui/src/redesign/universe/wizard, 2020-2023); F2023 = redesigned form
(managed/ui/src/redesign/features/universe/universe-form, default via enableNewUI: true, route /universes/create);
V2 = features-v2 create-universe wizard (route /create-universe, runtime-config gated). PASS = capability present;
GAP = not implemented (residual of sub-issue #6438).

target | tag / image (aarch64)                                  | tag SHA                                  | W2020 | F2023 default | #6437/#6379 live pricing summary | #6438 full-move confirm | #6440 auth toggles | #6439 review step (V2)         | #6438 leave-unsaved confirm | #6438 Reset Config confirm | image control
2024.2 | v2024.2.11.0 / 2024.2.11.0-b36-aarch64                 | 04544fd94fb4e50d05994784f444b075a4a5dfdb | absent| PASS          | PASS (UniverseResource.tsx:74)   | PASS                    | PASS               | n/a (V2 absent; F2023 single-page) | GAP                       | GAP                        | DIGEST OK, REVISION OK, NO_YBA
2025.1 | v2025.1.4.0 / 2025.1.4.0-b103-aarch64                  | 690ac4141e57d060fd834312f396643e8c41f260 | absent| PASS          | PASS                             | PASS                    | PASS               | n/a (V2 absent)                | GAP                         | GAP                        | DIGEST OK, REVISION OK, NO_YBA
2025.2 | v2025.2.6.0 / 2025.2.6.0-b111-aarch64                  | ae22ef35ff65de63303467d95ac3e6d2382feba9 | absent| PASS          | PASS                             | PASS                    | PASS               | PASS (V2 REVIEW step, price_per_hour) | GAP (V2 Cancel has no handler) | GAP                   | DIGEST OK, REVISION OK, NO_YBA
2026.1 | v2026.1.1.1 / 2026.1.1.1-b2-aarch64 (= latest)          | 3b010e021f6027744bf9e016b17e108a2eb24e44 | absent| PASS          | PASS                             | PASS                    | PASS               | PASS (V2 REVIEW step)          | GAP (V2 Cancel -> '/' no confirm) | GAP                  | DIGEST OK, REVISION OK, NO_YBA
master | preview v2.25.2.0 / 2.25.2.0-b359-aarch64; HEAD 52c3c2bf58 | 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d / 52c3c2bf5800aad3e9ae74f4d41b224868caac09 | absent | PASS | PASS | PASS | PASS | preview: n/a (V2 absent); HEAD: PASS | GAP (HEAD: Cancel/Close -> '/' no confirm) | GAP | DIGEST OK, REVISION OK, NO_YBA (preview image)

Per-target line citations (from yb4578-verify-ui.txt):
  UniverseForm.tsx renderSections :287 (all five tags); <CloudConfiguration> :290, <InstanceConfiguration> :291, <SecurityConfiguration> :294,
    <AdvancedConfiguration> :295, <GFlags> :298, <UserTags> :300; <UniverseResourceContainer> :196 (all five)
  UniverseResource.tsx pricePerHour :74 (all five); PlacementsFieldHelper.ts api.universeResource :280 (2024.2) / :283 (others);
    utils/api.ts /universe_resources :129 (2024.2) / :141 (others); v1.routes POST /universe_resources :256 (2024.2) / :257 (others)
  feature.js enableNewUI: true :29 (2024.2) / :28 (2025.1, 2025.2, 2.25.2.0) / :27 (2026.1, master HEAD)
  routes /universes/create -> UniverseNewView: routes.js:261 (2024.2, 2025.1, 2.25.2.0), routes.js:273 (2025.2), routes.jsx:280 (2026.1), routes.jsx:286 (master HEAD local)
  EditUniverse.tsx showFMModal :65 (2024.2) / :72 (others); FullMoveModal.tsx present on all five; en.json "Confirm Full Move Update" present on all five
  SecurityConfiguration.tsx <YSQLField :140 (2024.2) / :139 (2025.1, 2.25.2.0) / :150 (2025.2) / :151 (2026.1, master)
  UniverseDefinitionTaskParams.java enableYSQLAuth :478 / :484 / :590 / :630 / :480
  V2 create-universe: route routes.js:251 (2025.2), routes.jsx:252 (2026.1), routes.jsx:254 (master HEAD local); ReviewAndSummary.tsx price_per_hour :137 (2025.2) / :153 (2026.1) / :161 (master local)
  Residual: 0 files matching beforeunload|routerWillLeave|setRouteLeaveHook|useBlocker|<Prompt in universe-form or features-v2 create-universe on any tag;
    PlacementsField.tsx:242 `onClick={() => setValue(RESET_AZ_FIELD, true)}` (no modal) on master local; V2 CreateUniverse.tsx cancel: no onClick (2025.2), `window.location.href = '/'` (2026.1 :82, master HEAD)

Provenance (tag ancestry only; not runtime proof):
  828927b095 2020-07-20 [YW][#4578 - part 1]  -> first release tags v2.3.0.0 (also v2.3.1.0, ...)
  7e45cb5f42 2020-10-19 [YW][#4578] Create universe redesign (beta), adds "Try New UI" link -> v2.3.3.0, v2.4.0
  c1fda7ffe3 2020-11-23 [YW][#4578] Fixes and improvements (last commit referencing #4578) -> v2.5.1
  ac66da2c15 2023-01-30 [PLAT-7179] universe_code_revamp merge (F2023 added, enableNewUI: false) -> v2.17.2.0
  df6efc876f 2023-03-01 [PLAT-7545] enableNewUI default true -> v2.17.3.0 (stable line: v2.18.0.0)
  f185c02e87 2023-03-14 [PLAT-7712] Remove wizard flow for create and edit universe (W2020 deleted) -> v2.17.3.0 (stable: v2.18.0.0)
  4767d3f3cc 2025-04-22 [PLAT-16231] features-v2 create universe folder; 41481e8bfa 2025-07-03 V2 review-summary -> v2025.2.0.0
  W2020 present at v2.5.0 v2.5.3 v2.6.0 v2.8.0 v2.14.0.0 v2.16.0.0 v2.17.2.0; absent at v2.17.3.0 and every later tag incl. all five targets.


[PLAT-7179]: https://yugabyte.atlassian.net/browse/PLAT-7179?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4578-image-check.txt`):

# yb4578 image control; run at 2026-09-05T21:44:37Z via podman 5.8.6 (aarch64 host)
== 2024.2 docker.io/yugabytedb/yugabyte:2024.2.11.0-b36-aarch64 ==
2024.2 DIGEST OK sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
2024.2 yb-master --version: version 2024.2.11.0 build 36 revision 04544fd94fb4e50d05994784f444b075a4a5dfdb build_type RELEASE built at 18 Aug 2026 00:58:07 UTC
2024.2 REVISION OK 04544fd94fb4e50d05994784f444b075a4a5dfdb == tag SHA
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2024.2 NO_YBA: no yugaware/YBA/universe-form files in image
2024.2 www assets: bootstrap collapse.js d3.v2.js epoch.0.5.2.min.css epoch.0.5.2.min.js favicon.ico font-awesome images jquery-1.11.1.min.js libs logo.png yb.css 
2024.2 home: auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www 
== 2025.1 docker.io/yugabytedb/yugabyte:2025.1.4.0-b103-aarch64 ==
2025.1 DIGEST OK sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
2025.1 yb-master --version: version 2025.1.4.0 build 103 revision 690ac4141e57d060fd834312f396643e8c41f260 build_type RELEASE built at 02 Apr 2026 20:43:14 UTC
2025.1 REVISION OK 690ac4141e57d060fd834312f396643e8c41f260 == tag SHA
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2025.1 NO_YBA: no yugaware/YBA/universe-form files in image
2025.1 www assets: bootstrap collapse.js d3.v2.js epoch.0.5.2.min.css epoch.0.5.2.min.js favicon.ico font-awesome images jquery-1.11.1.min.js libs logo.png yb.css 
2025.1 home: auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www 
== 2025.2 docker.io/yugabytedb/yugabyte:2025.2.6.0-b111-aarch64 ==
2025.2 DIGEST OK sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
2025.2 yb-master --version: version 2025.2.6.0 build 111 revision ae22ef35ff65de63303467d95ac3e6d2382feba9 build_type RELEASE built at 02 Sep 2026 16:42:37 UTC
2025.2 REVISION OK ae22ef35ff65de63303467d95ac3e6d2382feba9 == tag SHA
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2025.2 NO_YBA: no yugaware/YBA/universe-form files in image
2025.2 www assets: bootstrap collapse.js d3.v2.js epoch.0.5.2.min.css epoch.0.5.2.min.js favicon.ico font-awesome images jquery-1.11.1.min.js libs logo.png yb.css 
2025.2 home: auto_flags.json bin controller cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib python share tools tserver tserver_flags.xml version_metadata.json www ybc-2.2.0.3-b22-el8-aarch64 
== 2026.1 docker.io/yugabytedb/yugabyte:2026.1.1.1-b2-aarch64 ==
2026.1 DIGEST OK sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
2026.1 yb-master --version: version 2026.1.1.1 build 2 revision 3b010e021f6027744bf9e016b17e108a2eb24e44 build_type RELEASE built at 14 Aug 2026 04:03:14 UTC
2026.1 REVISION OK 3b010e021f6027744bf9e016b17e108a2eb24e44 == tag SHA
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2026.1 NO_YBA: no yugaware/YBA/universe-form files in image
2026.1 www assets: bootstrap collapse.js d3.v2.js epoch.0.5.2.min.css epoch.0.5.2.min.js favicon.ico font-awesome images jquery-1.11.1.min.js libs logo.png yb.css 
2026.1 home: auto_flags.json bin controller cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib python share tools tserver tserver_flags.xml version_metadata.json www ybc-2.2.0.4-b5-el8-aarch64 
== master docker.io/yugabytedb/yugabyte:2.25.2.0-b359-aarch64 ==
master DIGEST OK sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
master yb-master --version: version 2.25.2.0 build 359 revision 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d build_type RELEASE built at 08 May 2025 19:05:35 UTC
master REVISION OK 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d == tag SHA
master DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
master DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
master DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
master DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
master NO_YBA: no yugaware/YBA/universe-form files in image
master www assets: bootstrap collapse.js d3.v2.js epoch.0.5.2.min.css epoch.0.5.2.min.js favicon.ico font-awesome images jquery-1.11.1.min.js libs logo.png yb.css 
master home: auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
